src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy

```diff
@@ -0,0 +1,86 @@
+package services.S3MHandlersAnalysis.datacollection
+
+import main.util.ProcessRunner
+import services.S3MHandlersAnalysis.Handlers
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+
+class S3MRunner {
+
+    static final Path S3M_PATH = Paths.get("src/services/S3MHandlersAnalysis/s3m.jar")
+
+    /**
+     * Runs S3M for each merge scenario and for each handler. Store the results at the same directory
+     * the merge scenario is located, in a directory for each handler.
+     *
+     * To extend the analysis for more handlers, check {@link #runHandlerVariants(Path, List < Handlers >)}
+     * @param mergeScenarios
+     * @param handlers
+     */
+    static void collectS3MResults(List<Path> mergeScenarios, List<Handlers> handlers) {
+        mergeScenarios.parallelStream()
+                .forEach(mergeScenario -> runHandlerVariants(mergeScenario, handlers))
+    }
+
+    private static void runHandlerVariants(Path mergeScenario, List<Handlers> handlers) {
+        Path leftFile = getInvolvedFile(mergeScenario, 'left')
+        Path baseFile = getInvolvedFile(mergeScenario, 'base')
+        Path rightFile = getInvolvedFile(mergeScenario, 'right')
+
+        // To extend the analysis for other handlers, clone and modify the following conditional.
+        if (handlers.contains(Handlers.Renaming)) {
+            runS3M(leftFile, baseFile, rightFile, 'CT.java', Handlers.Renaming, '-hmcrdov')
+            runS3M(leftFile, baseFile, rightFile, 'SF.java', Handlers.Renaming, '-r', 'SAFE')
+            runS3M(leftFile, baseFile, rightFile, 'MM.java', Handlers.Renaming, '-r', 'MERGE')
+            runS3M(leftFile, baseFile, rightFile, 'KB.java', Handlers.Renaming, '-r', 'BOTH')
+        }
+    }
+
+    private static void runS3M(Path leftFile, Path baseFile, Path rightFile, String outputFileName, Handlers handler, String... additionalParameters) {
+        Process S3M = ProcessRunner.startProcess(buildS3MProcess(leftFile, baseFile, rightFile, outputFileName, handler, additionalParameters))
+        S3M.getInputStream().eachLine {
+            //println it
```

Comment:
commented out code, should this loop be removed?

Reply:
Without "draining" all of the output, the process never finishes.

Reply:
running S3M.waitFor() woudn't wait for it to finish?

Reply:
Yes, but if the process has bufferized output, it won't finish. Here's a [link](https://stackoverflow.com/questions/5483830/process-waitfor-never-returns) about this.